test/unit/host_test.rb

```diff
@@ -3130,6 +3130,28 @@ def to_managed!
     end
   end
 
+  describe 'taxonomy scopes' do
+    test 'no_location overrides default scope' do
+      location = FactoryGirl.create(:location)
+      host = FactoryGirl.create(:host, :location => nil)
+      Location.stubs(:current).returns(location)
+
+      assert_nil Host.where(:id => host.id).first
+      assert_not_nil Host.no_location.where(:id => host.id).first
+      Location.unstub(:current)
```

Comment:
For some reason if I don't unstub the call, it messes with other tests (at least on my PC). I think it has something to do with the fact that it's a static method, so the stub is not cleaned properly.